.github/workflows/ci.yml

```diff
@@ -0,0 +1,45 @@
+# Syntax reference:
+# https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions
+
+name: Tests
+
+on:
+  push:
+    paths-ignore:
+      - '**.md'
+      - 'LICENSE'
+  pull_request:
+    paths-ignore:
+      - '**.md'
+      - 'LICENSE'
+
+jobs:
+  testing:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        cc: ['gcc', 'clang']
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install tgen dependencies
+        run: sudo apt install -y ${{ matrix.cc }} cmake libglib2.0-0 libglib2.0-dev libigraph0v5 libigraph0-dev
```

Comment:
Nit - "apt-get" is recommended over "apt" for scripting

Reply:
Note that I shamelessly copied `shadow/shadow/.github/workflows/build_shadow.yml` as a starting point, and `apt` is used there as well.

Reply:
Noted :). Yeah, I learned about this later